src/common/utils/sync-box/sync-box-injection-token.ts

```diff
@@ -4,12 +4,21 @@
  */
 import { getInjectionToken } from "@ogre-tools/injectable";
 import type { IComputedValue } from "mobx";
-import type { JsonValue } from "type-fest";
 
-export interface SyncBox<TValue extends JsonValue> {
+type AsJson<T> = T extends string | number | boolean | null
+  ? T
+  : T extends Function
+    ? never
+    : T extends Array<infer V>
+      ? AsJson<V>[]
+      : T extends object
+        ? { [K in keyof T]: AsJson<T[K]> }
+        : never;
```

Comment:
The previous definition used `TValue extends JsonValue` which doesn't work because that requires adding an index field to all the objects that we want to serialize. 

I can add some comments as to why this is necessary of you would like.

Reply:
Hm. I'm thinking if we really need anything else than JSON. It would make things a lot simpler. 
However, if you think it's needed, sure. We can simplify later if we so decide. :)